pandas/tests/indexes/timedeltas/test_arithmetic.py

```diff
@@ -25,26 +28,88 @@ def freq(request):
     return request.param
 
 
+def _raises_and_warns(op, left, right):
```

Comment:
name things properly if you must do this: assert_raises_type_error_and_performance_warning.

yes its pretty long but its readable. remember folks who have no history with this code will be reading it and have no idea unless its spelled our.